src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/RestSenderBase.cs

```diff
@@ -37,10 +37,12 @@ protected async Task SendAsync<T>(T model, string route)
                     this.logger.LogDebug("Sending content {0} to Uri {1}", request, publicationUri);
                     HttpResponseMessage httpResponseMessage = await client.PostAsync(publicationUri, request);
                     this.logger.LogDebug("Response: {0}", httpResponseMessage);
+                    return httpResponseMessage;
                 }
                 catch (Exception ex)
                 {
                     this.logger.LogError(ex, "Failed to send {0}", model);
+                    return null;
```

Comment:
Maybe not null but an actual response like 
new HttpResponseMessage() { StatusCode = HttpStatusCode.InternalServerError };
I am not sure of what you do with it after yet but ...

Reply:
Done.